src/components/Feed/Reactions.tsx

```diff
@@ -0,0 +1,91 @@
+import { Bomb, Fire, HandsPraying, Heart, Horse, Spinner } from "@phosphor-icons/react"
+import { useQuery } from "@tanstack/react-query";
+import { motion } from "framer-motion";
+import { forwardRef, useImperativeHandle, useRef, useState } from "react";
+import { api_reactions } from "../../shared/api";
+
+const reactionIcons = {
+  like: {
+    Icon: Heart,
+  },
+  unicorn: {
+    Icon: Horse,
+  },
+  exploding_head: {
+    Icon: Bomb,
+  },
+  raised_hands: {
+    Icon: HandsPraying,
+  },
+  fire: {
+    Icon: Fire,
+  },
+};
+
+export type ReactionsHandler = {
+    toggle: (articleId: number) => void;
+};
+
+export const Reactions = forwardRef<ReactionsHandler>((_, ref) => {
+  const [articleId, setArticleId] = useState<number | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const backdropRef = useRef<HTMLDivElement>(null);
+
+  const { data: reactions, isPending } = useQuery({
+    queryKey: ["reactions", articleId],
+    enabled: !!articleId,
+    queryFn: async () => {
+      const params = new URLSearchParams({
+        article_id: articleId?.toString() ?? "",
+      });
+      return (await api_reactions.get("/reactions?" + params)).data as Reactions;
+    },
+  });
+
+  useImperativeHandle(ref, () => ({
+    toggle: (articleId: number) => {
+      setArticleId(articleId);
+      setIsOpen(!isOpen);
+    },
+  }));
+
+  return (
+    <>
+      <div
+        ref={backdropRef}
+        className={`fixed inset-0 w-full h-full z-50 ${isOpen ? "" : "pointer-events-none"}`}
+      ></div>
+
+      <motion.div
+        className="absolute -bottom-8 right-14 px-8 py-4 min-w-[200px] bg-white rounded-2xl shadow-lg"
+        style={{ opacity: 0 }}
+        animate={{ opacity: isOpen ? 1 : 0 }}
```

Comment:
With 0 opacity, users cannot see the element, however users that use screen readers are still able to see it, as the element is still on the DOM. For that, we pass an aria prop specifying this element should not be seen.

```suggestion
        animate={{ opacity: isOpen ? 1 : 0 }}
        aria-hidden={!isOpen}
```

And i liked your approach 😀, it's better setting opacity 0 than display: none for animation purposes

Reply:
I have a question. In comments component has a similar code, but the animation is `translate: y`. Does this type of `aria` also work for this code?

![image](https://github.com/user-attachments/assets/c6bf5545-8ecc-48e5-aa04-ba9a4f186708)

Reply:
Ohhh great question! Yes it can be applied there too.
I didn't thought of it at first, thanks for noticing that, if it doesn't bother you, feel free to add it on your pr, or log an issue if you prefer